component/src/test/java/org/wso2/extension/siddhi/io/http/sink/HttpOAuthFailTestCase.java

```diff
@@ -0,0 +1,445 @@
+/*
+ *  Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.extension.siddhi.io.http.sink;
+
+import org.apache.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.extension.siddhi.io.http.sink.util.HttpOAuthListenerHandler;
+import org.wso2.extension.siddhi.map.xml.sinkmapper.XMLSinkMapper;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class HttpOAuthFailTestCase {
+    private static final Logger log = Logger.getLogger(HttpAuthTestCase.class);
+    public static final String CARBON_HOME = "carbon.home";
+    private AtomicInteger eventCount = new AtomicInteger(0);
+
+    /**
+     *Set carbon-home
+     */
+    private void setCarbonHome() {
+        Path carbonHome = Paths.get("");
+        carbonHome = Paths.get(carbonHome.toString(), "src", "test");
+        System.setProperty(CARBON_HOME, carbonHome.toString());
+        log.info("Carbon Home Absolute path set to: " + carbonHome.toAbsolutePath());
```

Comment:
Why is this step necessary?

Reply:
for https request, we need to set a keystore, the keystore is in the  <CARBON_HOME>/resources/security. so I create a method for getting the URL.